confluent_kafka/src/confluent_kafka.c

```diff
@@ -105,6 +122,14 @@ static PyMethodDef KafkaError_methods[] = {
 	  "  :rtype: str\n"
 	  "\n"
 	},
+        { "fatal", (PyCFunction)KafkaError_fatal, METH_NOARGS,
+          "  :returns: True if this a fatal error, else False.\n"
+          "  :rtype: bool\n"
+          "\n"
+        },
+        { "_test_raise_fatal", (PyCFunction)KafkaError_test_raise_fatal,
```

Comment:
does this really provide enough value in testability to warrant messying up the `KafkaError` class?

Reply:
ain't no other way to trigger a fatal error, short of introducing broker-side errors, so this is a must.
underscore prefixed methods are considered internal anyway.

Reply:
The fatal() functionality needs a test, which to me is more important than not messying up the class with a non-public and clearly internal method.
This is particularly important in this case where real fatal errors are really hard to trigger.

Reply:
what we really want is a mock broker, or proxy that can inject problems.

Reply:
a single underscore means protected by convention which does make it part of the public interface. better to make it __ -> private.

Reply:
I don't believe that's correct:
https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles

> _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose name starts with an underscore.
> ..
> __double_leading_underscore: when naming a class attribute, invokes name mangling (inside class FooBar, __boo becomes _FooBar__boo; see below).

> what we really want is a mock broker, or proxy that can inject problems.
surely, but I feel that might be outside the scope of this PR ;)